src/webapp/logic/sheetBuilder.ts

```diff
@@ -1094,24 +1095,94 @@ export class SheetBuilder {
         } else if (
             this.builder.useCodesForMetadata &&
             item?.code &&
-            ["organisationUnits", "dataElements", "options", "categoryOptions"].includes(item?.type)
+            item?.type &&
+            ["organisationUnits", "dataElements", "options", "categoryOptions"].includes(item.type)
         ) {
             return { name: item.code, description };
         } else {
-            return { name: selectedName ? item.displayShortName : name, description };
+            return { name: (selectedName ? item?.displayShortName : name) || "", description };
+        }
+    }
+
+    private getTranslatedMetadataName(props: { id: Maybe<string>; type: Maybe<string> }) {
+        const { id, type = "Metadata" } = props;
+        const metadata = this.builder.elementMetadata.get(id);
+        if (!metadata) {
+            console.error(`${type} not found: ${id}`);
+            return "";
+        }
+
+        const { name } = this.translate(metadata);
+        return name;
+    }
+
+    private buildCategoryOptionComboName(item: CategoryOptionCombo): string {
+        const categoryOptionIds = new Set(item.categoryOptions.map(({ id }) => id));
+
+        if (!categoryOptionIds.size) {
+            return "";
+        } else if (categoryOptionIds.size === 1) {
+            const [categoryOptionId] = categoryOptionIds;
+            return this.getTranslatedMetadataName({ id: categoryOptionId, type: "CategoryOption" });
+        } else {
+            return this.getMultipleCategoryOptionOptionsName(item, categoryOptionIds);
         }
     }
 
+    private getMultipleCategoryOptionOptionsName(item: CategoryOptionCombo, categoryOptionIds: Set<string>): string {
+        const { elementMetadata } = this.builder;
+
+        const categoryCombo = elementMetadata.get(item.categoryCombo.id) as TranslatableItem;
+        if (categoryCombo?.type !== "categoryCombos") {
+            console.error(`Category combo not found: ${item.categoryCombo.id}`);
+            return "";
+        }
+
+        const orderedCategoryOptions = this.getCategoryOrderedCategoryOption(categoryCombo, categoryOptionIds);
+
+        const isOrderedCategoryOptionsValid = orderedCategoryOptions.length === categoryOptionIds.size;
+        if (!isOrderedCategoryOptionsValid) {
+            console.error(
+                `Category options mismatch for combo ${item.categoryCombo.id}. Using CategoryOptionCombo options`
+            );
+        }
+
+        const categoryOptions = isOrderedCategoryOptionsValid ? orderedCategoryOptions : item.categoryOptions;
+
+        const options = categoryOptions.map(categoryOption => {
+            return this.getTranslatedMetadataName({ id: categoryOption.id, type: "CategoryOption" });
+        });
+        return options.join(", ");
+    }
+
+    private getCategoryOrderedCategoryOption(categoryCombo: CategoryCombo, categoryOptionIds: Set<string>) {
```

Comment:
The function returns a collection, so probably should be `getCategoryOrderedCategoryOptions`. 

Also, as best-practice, let's be explicit with the return type of (all) methods.